File: HW4/userid/DecoyDuck.cpp
#include <iostream>
#include "DecoyDuck.h"
#include "FlyNoWay.h"
#include "MuteQuack.h"
#include "DoesNotLayEggs.h"

DecoyDuck::DecoyDuck( ) {
   flyBehavior = new FlyNoWay( );
   quackBehavior = new MuteQuack( );
   eggBehavior = new DoesNotLayEggs( );
}

void DecoyDuck::display( ) {
   std::cout << "I’m a decoy duck" << std::endl;
}

File: HW14/huan1588/DecoyDuck.java
public class DecoyDuck extends Duck {

   public DecoyDuck( ) {
      flyBehavior = new FlyNoWay( );
      quackBehavior = new MuteQuack( );
      eggBehavior = new LaysEggBroody();
   }

   public void display( ) {
      System.out.println("I’m a decoy duck");
   }
}
